truffle/src/main/java/org/jruby/truffle/language/constants/LookupConstantWithLexicalScopeNode.java

```diff
@@ -64,4 +68,9 @@ protected boolean isVisible(RubyConstant constant) {
         return constant == null || constant.isVisibleTo(getContext(), lexicalScope, getModule());
     }
 
+    @TruffleBoundary
+    private void warnDeprecatedConstant(String name) {
+        getContext().getJRubyRuntime().getWarnings().warn("constant " + name + " is deprecated");
+    }
```

Comment:
Maybe a `WarnNode` that allows varargs arguments so it handle the String concatenation nicely on its side.